src/main/haskell/kore/test/Test/Kore/Step/Simplification/Exists.hs

```diff
@@ -0,0 +1,320 @@
+module Test.Kore.Step.Simplification.Exists
+    ( test_existsSimplification
+    ) where
+
+import Test.Tasty
+       ( TestTree )
+import Test.Tasty.HUnit
+       ( testCase )
+
+import qualified Data.Map as Map
+import           Data.Reflection
+                 ( Given, give )
+
+import           Kore.AST.Common
+                 ( AstLocation (..), Exists (..), Id (..), Sort (..),
+                 SortActual (..), Variable )
+import           Kore.AST.MetaOrObject
+import           Kore.ASTUtils.SmartConstructors
+                 ( mkAnd, mkApp, mkEquals, mkExists, mkTop, mkVar )
+import qualified Kore.Error
+import           Kore.IndexedModule.MetadataTools
+                 ( MetadataTools, SortTools )
+import           Kore.Predicate.Predicate
+                 ( makeCeilPredicate, makeEqualsPredicate, makeExistsPredicate,
+                 makeTruePredicate )
+import           Kore.Step.ExpandedPattern
+                 ( CommonExpandedPattern, ExpandedPattern (ExpandedPattern) )
+import qualified Kore.Step.ExpandedPattern as ExpandedPattern
+                 ( ExpandedPattern (..), bottom, top )
+import           Kore.Step.OrOfExpandedPattern
+                 ( CommonOrOfExpandedPattern, OrOfExpandedPattern )
+import qualified Kore.Step.OrOfExpandedPattern as OrOfExpandedPattern
+                 ( make )
+import           Kore.Step.Simplification.Data
+                 ( evalSimplifier )
+import qualified Kore.Step.Simplification.Exists as Exists
+                 ( makeEvaluate, simplify )
+import qualified Kore.Step.Simplification.Simplifier as Simplifier
+                 ( create )
+import           Kore.Step.StepperAttributes
+                 ( StepperAttributes )
+
+import           Test.Kore.Comparators ()
+import qualified Test.Kore.IndexedModule.MockMetadataTools as Mock
+                 ( makeMetadataTools, makeSortTools )
+import qualified Test.Kore.Step.MockSymbols as Mock
+import           Test.Tasty.HUnit.Extensions
+
+test_existsSimplification :: [TestTree]
+test_existsSimplification = give mockSortTools
+    [ testCase "Exists - or distribution"
+        -- exists(a or b) = exists(a) or exists(b)
+        (assertEqualWithExplanation ""
+            (OrOfExpandedPattern.make
+                [ ExpandedPattern
+                    { term = mkExists Mock.x something1OfX
+                    , predicate = makeTruePredicate
+                    , substitution = []
+                    }
+                , ExpandedPattern
+                    { term = mkExists Mock.x something2OfX
+                    , predicate = makeTruePredicate
+                    , substitution = []
+                    }
+                ]
+            )
+            (evaluate mockMetadataTools
+                (makeExists
+                    Mock.x
+                    [something1OfXExpanded, something2OfXExpanded]
+                )
+            )
+        )
+    , testCase "Exists - bool operations"
+        (do
```

Comment:
Why are these in a do?

Reply:
It seemed to make sense to group them together, since they were fairly similar